sierralocal/main.py

```diff
@@ -51,10 +52,34 @@ def scorefile(input_file, algorithm, do_subtype=False, program='post'):
     aligner = NucAminoAligner(algorithm, program=program)
     result = aligner.align_file(input_file, program=program)
 
+    # hold all NNN sequences and void them when scoring drugs
+    ambiguous = {}  # {sequence name: {gene : set(positions of NNN)}}
+    names = []
+    gene_order = {}
+
+
     print('Aligned ' + input_file)
     sequence_headers, file_genes, file_mutations, file_trims, subtypes = \
         aligner.get_mutations(result, do_subtype=do_subtype)
 
+
+    for ind, gene in enumerate(file_genes):
+        seq_n = sequence_headers[ind]
+        gene_order[seq_n] = []
+        
+        for ind2, protein in enumerate(gene):
+            gene_order[seq_n].append(protein[0])
+
+    for ind, sequence in enumerate(file_mutations):
+        seq_n = sequence_headers[ind]
+        ambiguous[seq_n] = {}
+        for ind2, muts in enumerate(sequence):
+            ambiguous[seq_n][gene_order[seq_n][ind2]] = set()
+            for position, AA in muts.items():
+                if len(AA[1]) == 21: # checks if was NNN
```

Comment:
why is `21` hard-coded here?  is it one past the number of amino acids?  Need a bit more explanation here

Reply:
The last one is * for stop codon I believe. I could try to find a more elegant solution by looking further upstream